vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/PublicSearchManager.java

```diff
@@ -0,0 +1,144 @@
+        Page<IModule> modulePage = moduleManager.findByNameOrDescription(requestedPageForModule, searchTerm);
+        int totalModulePage = modulePage.getTotalPages();
+        if (page > totalModulePage) {
+            totalModulePage = totalModulePage == 0 ? 1 : totalModulePage;
+            requestedPageForModule = PageRequest.of(totalModulePage - 1, pageSize);
+            modulePage = moduleManager.findByNameOrDescription(requestedPageForModule, searchTerm);
+        }
+        return modulePage;
+    }
+
+    /**
+     * Method to return the requested slides whose name or description contains the
+     * search string
+     * 
+     * @param searchTerm string which has been searched
+     * @param page.      The page number starts from 1.if page<1, 1st page is
+     *                   returned, if page>total pages,last page is returned
+     * @return set of slides whose name or description contains the search string in
+     *         the requested page.
+     */
+    @Override
+    public Page<ISlide> searchInSlides(String searchTerm, int page) {
+        if (page < 1) {
+            page = 1;
+        }
+        Pageable requestedPageForSlide = PageRequest.of(page - 1, pageSize);
+        Page<ISlide> slidePage = slideManager.findByNameOrDescription(requestedPageForSlide, searchTerm);
+        int totalSlidePage = slidePage.getTotalPages();
+        if (page > totalSlidePage) {
+            totalSlidePage = totalSlidePage == 0 ? 1 : totalSlidePage;
+            requestedPageForSlide = PageRequest.of(totalSlidePage - 1, pageSize);
+            slidePage = slideManager.findByNameOrDescription(requestedPageForSlide, searchTerm);
+        }
+        return slidePage;
+    }
+
+    /**
+     * Method to return the requested slides whose text blocks contains the search
+     * string
+     * 
+     * @param searchTerm string which has been searched
+     * @param page.      The page number starts from 1.if page<1, 1st page is
+     *                   returned, if page>total pages,last page is returned
+     * @return list of slides whose text blocks contains the search string in the
+     *         requested page.
+     */
+    @Override
+    public Page<ISlide> searchInSlideTexts(String searchTerm, int page) {
+        if (page < 1) {
+            page = 1;
+        }
+        Pageable requestedPageForSlideText = PageRequest.of(page - 1, pageSize);
+        Page<ISlide> slidetextPage = textContentBlockRepo.findWithNameOrDescription(requestedPageForSlideText, searchTerm);
+        int totalSlideTextPage = slidetextPage.getTotalPages();
+        if (page > totalSlideTextPage) {
+            totalSlideTextPage = totalSlideTextPage == 0 ? 1 : totalSlideTextPage;
+            requestedPageForSlideText = PageRequest.of(totalSlideTextPage - 1, pageSize);
+            slidetextPage = textContentBlockRepo.findWithNameOrDescription(requestedPageForSlideText, searchTerm);
+        }
+        return slidetextPage;
+    }
+
+}
```

Comment:
this class has a lot of overlapping code with the staffsearchmanager, doesn't it? those two should be consolidated to make it DRY.